### DISCUSSION TITLE
Feedback wanted on NumericField, editor 'Spinner'

### DISCUSSION THREAD

**Discussion by netwavebe** [OrchardCMS/OrchardCore#7387] [Ideas]
While working on NumericField, I noticed the 'spinner' editor has some issues. It's based on jQuery's spinner().

Layout of this field is not in line with the other fields (double border and different width):

(NumericField01 is the default editor, NumericField05 is the spinner editor).

> ![image](https://user-images.githubusercontent.com/3008547/97111174-9b0f8080-16dd-11eb-940d-1e7f38824188.png)

It also has responsive issues:

> ![image](https://user-images.githubusercontent.com/3008547/97111187-b5e1f500-16dd-11eb-953f-a0752bdca21e.png)

I would like to change it to only use Bootstrap components, maybe something like this:

> ![image](https://user-images.githubusercontent.com/3008547/97111395-ba5add80-16de-11eb-9951-c532c36fed03.png)

This would solve both the styling & responsive issues.

What's the group's take on this? Should I go and implement this?

**Comment by Skrypt** [comment]
Without looking at the code it should be using a btn-group.

**Reply by netwavebe** [reply]
Html looks good to me:

https://github.com/OrchardCMS/OrchardCore/blob/dev/src/OrchardCore.Modules/OrchardCore.ContentFields/Views/NumericField-Spinner.Edit.cshtml#L17

It"s the jQuery plugin that breaks things by applying extra classes to the input-element.

**Reply by Skrypt** [reply]
`input-goup` instead of `btn-group` yep it's fine. Have you tried with a display-inline or nowrap of some sort?

basically it should be something like : 

```
display: inline-block;  == d-*-inline bs4 class
white-space: nowrap; == text-nowrap bs4 class
```

If it doesn't work I'll take a closer look.

**Reply by netwavebe** [reply]
I'd rather not try to fix the jQuery plugin, but instead replace it with pure Bootstrap components and some javascript. Like the last screenshot shows.

The jQuery plugin breaks responsive behaviour, applies unwanted styling and the up/down buttons are too small imho.

Hence my question, I'd like to know if people think my proposition is okay and I should make a PR to change this.

**Reply by Skrypt** [reply]
If we can remove JQuery out of the picture in favor of using Bootstrap components I'm all in 😉 
One less dependency is always a good thing.

**Reply by netwavebe** [reply]
My idea exactly!

**Comment by netwavebe** [comment]
Also related to https://github.com/OrchardCMS/OrchardCore/issues/2266

**Comment by netwavebe** [comment]
Fixed in https://github.com/OrchardCMS/OrchardCore/pull/7421
